front_end/content.py

```diff
@@ -1407,18 +1417,24 @@ def save_course(self, course_basics, course_details):
         return course_basics["id"]
 
     def save_assignment(self, assignment_basics, assignment_details):
+        # clean and join allowed_ip_addresses
+        assignment_details["allowed_ip_addresses"][:] = [x for x in assignment_details["allowed_ip_addresses"] if x != "" and x != ","]
+        allowed_ip_addresses_string = ",".join(assignment_details["allowed_ip_addresses"])
```

Comment:
I'd rather you put the modified string back in the assignment_details dictionary (using the same `allowed_ip_addresses` key rather than create a new variable.